### ISSUE TITLE
Mapfile Plotting Improvements

### ISSUE THREAD

**Issue by asreimer** [vtsuperdarn/davitpy#167]
There are a couple of issues with the current mapfile plotting procedures (code is found in the [davitpy/pydarn/plotting/plotMapGrd.py](https://github.com/vtsuperdarn/davitpy/blob/develop/davitpy/pydarn/plotting/plotMapGrd.py) file). Here's a summary of things that need to be done:

1) Proper MLT plots (noon should be at the top of the plot, mlon labels should be converted to hours, etc.)
2) `__init__` routine for MapConv needs to be more flexible
    a) shouldn't force a user to load both a map file and a grid file
    b) should accept a fileName and/or a mapPtr object instead the current hardcoded remote file fetching
3) Add a function that simply creates a convection map plot with one line of code (like we have in rti and fan plotting)
4) Possibly more minor fixes and some minor refactoring of the code (polish)

For example, to crudely address 2), here's what I just had to do and recommend to @GPerry306 so that he could plot from a local mapfile. 

Add `,mapPtr=None` to the `__init__` on line 53. And replace lines 85-88 with:

```
if mapPtr is None:
    grdPtr = sdDataOpen(startTime, hemi, eTime=endTime, fileType='grdex')
    self.grdData = grdPtr.readRec()
    mapPtr = sdDataOpen(startTime, hemi, eTime=endTime, fileType='mapex')
    self.mapData = mapPtr.readRec()
else:
    self.mapData = mapPtr.readRec()
```

To illustrate the need for 3) (and a bit of 2)), here's what currently needs to be done to plot a convection map from a local file:

```
#Create a maplotlib figure and axis object
from matplotlib import pyplot
fig = pyplot.figure()
ax = fig.add_subplot(111)

#Open a data file via path to file (use fileName optional arg)
mapPtr = pydarn.sdio.sdDataOpen(datetime(2015,6,25),hemi='north',fileType='map',fileName='20150625.map')

#Create a blank map object in mlt coords to use for plotting (this uses matplotlib's basemap) using the start time for the map data we want to plot.
mObj = utils.plotUtils.mapObj(datetime=datetime(2015,6,25,2),boundinglat=50.,gridLabels=True,coords='mlt')

#Now we pass the mapPtr, mObj, and ax to the MapConv function in plotMapGrd. Choose the time that you want to plot for, for example, for the start of the day I just used datetime(2015,6,25):
mapDatObj = pydarn.plotting.plotMapGrd.MapConv(datetime(2015,6,25), mObj, ax, mapPtr=mapPtr)

#Now we add the fitted velocities, the map contours, and the HMB
mapDatObj.overlayMapFitVel()
mapDatObj.overlayCnvCntrs()
mapDatObj.overlayHMB()

#Finally, show the figure!
fig.show()
```

This should be easy to fix, just needs someone to work on it.

**Comment by aburrell** [comment]
I was just looking at this, and it also needs to accept “password” as a kwarg, allowing downloads at servers other than VT.

**Comment by drmike2017** [comment]
Hello Everyone,

   In order to support my post-doc research, I have been making some changes to the DaViTpy software here in Saskatoon. Specifically, I have been making changes to the convection map modules in order to address this open "enhancements" thread. I have been able to perform much of the tasks specified, including converting the MLT coordinates to the 24-hour convention, placing 12h at the top of the plot, and creating a one-line command to produce a convection map. Using Ashton's "rough code" above, I have also been able to access local map files, thus finally avoiding the somewhat cryptic exception message when not using the VT server.

   I have also made some minor cosmetic changes to the convection map, such as placing the color bar a little further away from the plot frame borders, allowing all of the meridian labels (on all sides of the frame) to be shown and placing the coordinate message (MLT in the example image) above and away from the meridian labels. Finally, I removed the continent fill and used only light boundaries for the continents in order to avoid any obscuration of convection data and the contour lines.

   I ended up modifying the plotMapGrd and plotUtils modules in order to accomplish this. I also created a small module containing a method called "mappy" to contain all of the steps shown in Ashton's first message above. Once I clean up the code, I will be offering it as a pull request, likely on Monday. In the meantime, I have posted an example image of the output to see if anyone has any suggestions or proposed additions.

   As a PS, the matplotlib version installed on our Maxwell server is 1.5.1 and so it is affected by the ticker glitch when attempting to plot LinearLocator contour lines. Instead, I used the AutoLocator (default) until a more up to date version of matplotlib is installed on Maxwell.

![cmap - 060113 - 0122-24 final](https://user-images.githubusercontent.com/33164719/33783977-98a917bc-dc24-11e7-9c57-bb2bda1f3b54.jpg)

Thanks.

Mike.
